bug: start and end time inconsistent and duplicates missing

js3110 opened the pull request:
## Issue

Closes #214 

## Description

During development, the derivation of start and end time has swapped occasionally between being the time from first dose, or the time from most recent dose. Due to this, there are some sections later on in the code that are inconsistent- ie some parts assume that it is time from first dose, and others that it is time from mrd. 
Lambda.z.ix is also not correct due to the start times and also the duplicates are missing.

## Definition of Done

- [x] start and end time used as correct time for calculations
- [x] Plots show the correct time, and lambda.z.ix correctly calculated

## Contributor checklist
- [x] Code passes lintr checks
- [x] Code passes all unit tests
- [x] New logic covered by unit tests
- [x] New logic is documented
- [x] Package version is incremented

## Notes to reviewer

m-kolomanski reviewed (approved):
LGTM.

Yes, the package version should be bumped with every change to the code.

This might be skipped when PR only involves some work outside of the package and app itself (CI updates, vignettes/articles, README or CONTRIBUTING updates etc).

Gero1999 reviewed (changes requested):
hey @js3110 thanks for checking this, just a comment on it

js3110 commented:
@Gero1999 as discussed I changed it so that the results show start and end time from first dose.
I created temporary variables start_dose and end_dose for the parts of the calculations that require it to be from recent dose.

I also modified the slope plots so they show the time from recent dose, as for the exploration and manual slopes selection this is better as they can directly see tmax and lambda.z.time.first without having to do any mental maths.

I hope this makes sense to you now !

@m-kolomanski if you could quickly re review as well, just for safety 😄

m-kolomanski reviewed (approved):
TMLG!
